code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -82,6 +82,16 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
 
 	return ..()
 
+/datum/antagonist/traitor/select_organisation()
+	var/chaos = pick(prob(ORG_PROB_HUNTER);ORG_CHAOS_HUNTER, prob(ORG_PROB_MILD);ORG_CHAOS_MILD, prob(ORG_PROB_AVERAGE);ORG_CHAOS_AVERAGE, prob(ORG_PROB_HIJACK);ORG_CHAOS_HIJACK)
+	var/org_list = shuffle(subtypesof(/datum/antag_org/syndicate))
+	for(var/org_type in org_list)
+		var/datum/antag_org/org = new org_type(src)
+		if(org.chaos_level == chaos)
+			organisation = org
+			break
+		qdel(org)
```

Comment:
can't you check the chaos level of the org using `initial()` without have to new() that type?

Reply:
I'm not sure how I'd go about doing that, do you have an example on the codebase?

Reply:
```dm
	for(var/path in subtypesof(/datum/keybinding))
		var/datum/keybinding/D = path
		if(initial(D.name))
			GLOB.keybindings += new path()
```

Reply:
Thanks! Done and tested, works